storageprovider/csp/container_storage_provider.go

```diff
@@ -484,6 +495,11 @@ func (provider *ContainerStorageProvider) GetVolume(id string) (*model.Volume, e
 			ResponseError: &errorResponse,
 		},
 	)
+
+	if status == http.StatusOK {
+		return dataWrapper.Data.(*model.Volume), nil
+	}
+
```

Comment:
Since we are returning here on success, the end statement of function should be `return nil, err`?

Reply:
yup, ideally we should never reach there but will modify the end statement